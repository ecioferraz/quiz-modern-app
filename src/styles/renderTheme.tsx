import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export default function renderTheme(children: ReactNode): RenderResult {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
