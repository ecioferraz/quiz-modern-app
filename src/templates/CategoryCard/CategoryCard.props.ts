type CategoryCardProps = {
  background: string;
  category: string;
  iconSrc: string;
  rating?: boolean;
};

export default CategoryCardProps;
