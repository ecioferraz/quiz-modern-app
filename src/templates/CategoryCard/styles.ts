import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    align-items: center;
    background: #FFFFFF;
    box-shadow: 10px 24px 54px rgba(51, 51, 51, 0.03);
    border-radius: 10px;
    display: flex;
    justify-content: space-between;
    margin: ${theme.spacings.small} 0;

    > * {
      margin: ${theme.spacings.small};
      margin-right: 1.9rem;
    }
  `}
`;

export const CategoryContainer = styled.section`
  ${({ theme }) => css`
    align-items: center;
    display: flex;
    gap: 1.6rem;

    > section:has(img) {
      align-items: center;
      border-radius: 10px;
      box-shadow: 10px 24px 54px rgba(51, 51, 51, 0.05);
      display: flex;
      height: 50px;
      justify-content: center;
      width: 50px;
    }
  `}
`;

export const CategoryDescriptionContainer = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: 0.8rem;

    h2, p {
      margin: 0;
    }

    h2 {
      font-size: 1.4rem;
      font-weight: ${theme.font.weight.semiBold};
    }

    p {
      color: ${theme.font.color.tertiaryColor};
      font-family: ${theme.font.family.secondaryFont};
      font-size: 1.2rem;
      font-weight: ${theme.font.weight.regular};
    }
  `}
`;

export const RatingContainer = styled.section`
  ${({ theme }) => css`
    align-items: center;
    display: flex;
    gap: 0.6rem;

    p {
      font-size: 1.4rem;
    }
  `}
`;