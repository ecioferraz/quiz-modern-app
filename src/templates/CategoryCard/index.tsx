import SvgCard from '../../components/SvgCard';
import TextCard from '../../components/TextCard';
import CategoryCardProps from './CategoryCard.props';
import * as Styled from './styles';

export default function CategoryCard({
  background,
  category,
  iconSrc,
  rating = false,
}: CategoryCardProps) {
  return (
    <Styled.Container>
      <Styled.CategoryContainer>
        <SvgCard
          alt={`${category} icon`}
          background={background}
          src={iconSrc}
        />
        <Styled.CategoryDescriptionContainer>
          <TextCard as="h2" size="small">
            {category}
          </TextCard>
          <TextCard as="p" size="small">
            10 Questions
          </TextCard>
        </Styled.CategoryDescriptionContainer>
      </Styled.CategoryContainer>
      {rating && (
        <Styled.RatingContainer>
          <SvgCard alt="Rating star" src="assets/icons/star.svg" />
          <TextCard as="p" size="small">
            4.8
          </TextCard>
        </Styled.RatingContainer>
      )}
    </Styled.Container>
  );
}
