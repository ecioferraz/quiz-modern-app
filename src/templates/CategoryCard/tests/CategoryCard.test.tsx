import renderTheme from '../../../styles/renderTheme';
import CategoryCard from '..';

describe('<CategoryCard />', () => {
  it('should render <CategoryCard /> with rating', () => {
    const { container } = renderTheme(
      <CategoryCard
        background="#FFE3BE"
        category="Technology"
        iconSrc="assets/icons/desktop.svg"
        rating
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render <CategoryCard /> without rating', () => {
    const { container } = renderTheme(
      <CategoryCard
        background="#FFE3BE"
        category="Technology"
        iconSrc="assets/icons/desktop.svg"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
