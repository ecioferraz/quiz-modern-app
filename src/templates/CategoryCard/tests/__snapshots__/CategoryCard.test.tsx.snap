// Vitest Snapshot v1

exports[`<CategoryCard /> > should render <CategoryCard /> with rating 1`] = `
.c2 {
  background: #FFE3BE;
}

.c4 {
  font-size: 1.6rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.03);
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.6rem 0;
}

.c0 > * {
  margin: 1.6rem;
  margin-right: 1.9rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c1 > section:has(img) {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c3 h2,
.c3 p {
  margin: 0;
}

.c3 h2 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c3 p {
  color: #999999;
  font-size: 1.2rem;
  font-weight: 400;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.6rem;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <section
        class="c2"
      >
        <img
          alt="Technology icon"
          src="assets/icons/desktop.svg"
        />
      </section>
      <section
        class="c3"
      >
        <h2
          class="c4"
        >
          Technology
        </h2>
        <p
          class="c4"
        >
          10 Questions
        </p>
      </section>
    </section>
    <section
      class="c5"
    >
      <section
        class=""
      >
        <img
          alt="Rating star"
          src="assets/icons/star.svg"
        />
      </section>
      <p
        class="c4"
      >
        4.8
      </p>
    </section>
  </section>
</div>
`;

exports[`<CategoryCard /> > should render <CategoryCard /> without rating 1`] = `
.c2 {
  background: #FFE3BE;
}

.c4 {
  font-size: 1.6rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.03);
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.6rem 0;
}

.c0 > * {
  margin: 1.6rem;
  margin-right: 1.9rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c1 > section:has(img) {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c3 h2,
.c3 p {
  margin: 0;
}

.c3 h2 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c3 p {
  color: #999999;
  font-size: 1.2rem;
  font-weight: 400;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <section
        class="c2"
      >
        <img
          alt="Technology icon"
          src="assets/icons/desktop.svg"
        />
      </section>
      <section
        class="c3"
      >
        <h2
          class="c4"
        >
          Technology
        </h2>
        <p
          class="c4"
        >
          10 Questions
        </p>
      </section>
    </section>
  </section>
</div>
`;
